components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -134,24 +147,48 @@ static void psa_mac_operation(void)
                 }
 
                 case PSA_MAC_UPDATE: {
-                    uint8_t *input_ptr = mbedtls_calloc(1, msg.in_size[1]);
-                    if (input_ptr == NULL) {
-                        status = PSA_ERROR_INSUFFICIENT_MEMORY;
-                        break;
-                    }
 
-                    bytes_read = psa_read(msg.handle, 1, input_ptr,
-                                          msg.in_size[1]);
+                    size_t data_remaining = msg.in_size[1];
+                    size_t size_to_read = 0;
+                    psa_status_t failure_status = PSA_SUCCESS;
 
-                    if (bytes_read != msg.in_size[1]) {
-                        SPM_PANIC("SPM read length mismatch");
+                    if (input_buffer_ptr == NULL)
+                    {
+                        input_buffer_ptr = mbedtls_calloc(1, MAX_DATA_CHUNK_SIZE);
+                        if (input_buffer_ptr == NULL) {
+                            status = PSA_ERROR_INSUFFICIENT_MEMORY;
+                            break;
+                        }
                     }
 
-                    status = psa_mac_update(msg.rhandle,
-                                            input_ptr,
-                                            msg.in_size[1]);
+                    while (data_remaining > 0)
```

Comment:
why do we want to keep reading once one of the Crypto operation failed?
```suggestion
                    while ((data_remaining > 0) && (PSA_SUCCESS == failure_status))
```

Reply:
there is a reason why I intentionally processed the entire payload even after an error is encountered.
I want the behavior in terms of timing invariance to be as close between the IPC and non-IPC versions of the function. since the non-IPC would pass all the data to the function, the non-IPC one should probably do the same.

Reply:
I understand the (near) constant time reasoning, but can an attacker exploit `psa_xxx_update` functions using timing attacks?

Reply:
I'm not sure an actual attack is relevant here but I want the IPC behavior to be as similar as possible to the NON-IPC one while still saving memory.